### ISSUE TITLE
Possibly create WeChat interface

### ISSUE THREAD

**Issue by jywarren** [publiclab/plotsbot#49]
Not very high priority, but an idea:

https://www.npmjs.com/package/wechat

https://github.com/wechat-bots/wechat-bot

**Comment by ananyo2012** [comment]
Yes PublicLab has a China Community as per I remember. And WeChat is widely
used in China.

+1 to the idea.

On Jun 28, 2017 12:58 AM, "Jeffrey Warren" <notifications@github.com> wrote:

Not very high priority, but an idea:

https://www.npmjs.com/package/wechat

https://github.com/wechat-bots/wechat-bot

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/publiclab/plotsbot/issues/49>, or mute the thread

**Comment by ebarry** [comment]
@shantler what do you think?

--

+1 336-269-1539 / @lizbarry <http://twitter.com/lizbarry> / lizbarry.net

**Comment by ryzokuken** [comment]
@ebarry They weren't mentioned. Are you sure they are on Github? They must be using a different username in that case.

**Comment by ebarry** [comment]
i fixed it! take a look at this WeChat idea @shanlter

**Comment by shanlter** [comment]
100% for this idea!! I can share the wechat login to someone, maybe @jywarren ? So you could play with it when you have time?

**Comment by ryzokuken** [comment]
@shanlter I would love to know more about it... Are specific credentials required to join the WeChat room? Isn't it open for everyone? If it is, the bot could just join the room. Will look into WeChat more extensively over the next few days.

**Comment by shanlter** [comment]
@ryzokuken "WeChat room" you mean WeChat official account? There are many types of WeChat official account, each type ask for different credentials, correspondingly has different function. Public Lab's Wechat account is: PublicLabChina.  it's a basic type of official account. But still has many build-in functions. more info around the official account here: https://admin.wechat.com

**Comment by ryzokuken** [comment]
@shanlter It would probably take me some time to get used to WeChat's terminology. In all the services I have used in the past, users have personal accounts and are supposed to join a group, team, or a chatroom. (eg: Publiclab's IRC chatroom, Matrix Group or Slack Team)

**Comment by ryzokuken** [comment]
@jywarren let's discuss this and make this one in the next PR? I am not used to using WeChat and I am a little wary of its policies so I might need a little help here. Reading https://github.com/node-webot/wechat/blob/master/README.en.md at the moment.

**Comment by ryzokuken** [comment]
Okay, upon on further inspection, it seems that it uses express to recieve requests and respond to them. The question arises: Do we have a static URL for the bot that it will recieve requests on? Also, that might not qualify as a legit interface because we cannot send messages back and forth, just reply to messages (which sounds okay-ish for a bot, but that's not how its configured to work, so that might mean a complete rewrite of the interfaces spec. I will look into more WeChat documentation, but it seems everything is in Chinese :stuck_out_tongue:

**Comment by shanlter** [comment]
@ryzokuken WeChat has personal account, group, official account, and it also function like twitter, paypal, uber, etc, but I don't think WeChat has chatroom. I am not quite understand what you want to build with wechat, but I'd happy to translate if in need.

**Comment by ryzokuken** [comment]
@shanlter Oh. Now I understand what you mean by Publiclab's official account. No, we should keep the Publiclab official account as-is. We should make another account named "plotsbot" and make it join the publiclab group. Then, the people on the group would be able to chat to the chatbot and make use of its features. Maybe now I'm clearer? I'd love to discuss the possibilites with you.

**Comment by ebarry** [comment]
@lilliankorinek check this out! (via @shelbyfirebaugh)

**Comment by lilliankorinek** [comment]
@ebarry Thank you! Shelby was just telling me about this. This is definitely up our alley!
